cluster/juju/layers/kubernetes-master/templates/service-always-restart.conf

```diff
@@ -0,0 +1,3 @@
+[Service]
+Restart=Always
+RestartSec=15
```

Comment:
I would propose the following instead:
```
[Unit]
StartLimitIntervalSec=0

[Service]
RestartSec=1
```

`StartLimitIntervalSec=0` disables start rate limiting. That's what we need to prevent systemd from "giving up".
`RestartSec=1` just because the default of 100ms is probably a little too rapid for our purposes. 1 second seems plenty slow but I'm open to other opinions.
`Restart=Always` probably won't help us since it's just about the exit code. I do think it's nice to have just in case we get an unexpected exit code 0, but we can actually put this one in the snapcraft.yaml instead, like we did with kubelet ([link](https://github.com/juju-solutions/release/blob/697052e40ed63c61a035b83c037f6fa86c2a705c/snap/kubelet.yaml#L19)).

Docs: [systemd.unit](https://www.freedesktop.org/software/systemd/man/systemd.unit.html), [systemd.service](https://www.freedesktop.org/software/systemd/man/systemd.service.html)